From c473e97e0fff4a5d51feaf8418d624fd9a82eb72 Mon Sep 17 00:00:00 2001
From: Oswald Buddenhagen <oswald.buddenhagen@gmx.de>
Date: Thu, 8 Oct 2020 13:24:59 +0200
Subject: [PATCH 2/7] do not avoid big stack allocations

these are cheap and unproblematic (we aren't in the kernel here).

amends ff0fc17a.
---
 src/subshell/common.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/subshell/common.c b/src/subshell/common.c
index 33c16fdb3..1d422999e 100644
--- a/src/subshell/common.c
+++ b/src/subshell/common.c
@@ -509,9 +509,8 @@ synchronize (void)
 static gboolean
 read_command_line_buffer (gboolean test_mode)
 {
-    /* Make static to avoid allocation of large buffers in the stack each time */
-    static char subshell_command_buffer[BUF_LARGE];
-    static char subshell_cursor_buffer[BUF_SMALL];
+    char subshell_command_buffer[BUF_LARGE];
+    char subshell_cursor_buffer[BUF_SMALL];
 
     fd_set read_set;
     int i;
-- 
2.28.0.1.g7ba5ae0684

